sympy/calculus/util.py

```diff
@@ -253,6 +253,79 @@ def elm_domain(expr, intrvl):
         return _domain
 
 
+def periodicity(f, symbol):
+    """Returns the period of a given function, if possible.
+
+    Examples
+    ========
+    >>> from sympy import Symbol, sin, cos
+    >>> from sympy.calculus.util import periodicity
+    >>> x = Symbol('x')
+    >>> f = sin(x) + sin(2*x) + sin(3*x)
+    >>> periodicity(f, x)
+    2*pi
+    >>> periodicity(sin(x)*cos(x), x)
+    pi
+    """
+    from sympy import simplify, lcm_list
+    from sympy.functions.elementary.trigonometric import TrigonometricFunction
+
+    orig_f = f
+    f = simplify(orig_f)
+    period = None
+
+    if isinstance(f, TrigonometricFunction):
+        period = f.period(symbol)
+
+    if f.is_Mul:
+        _, g = f.as_independent(symbol, as_Add=False)
+        if isinstance(g, TrigonometricFunction):
+            period = g.period(symbol)
+
+    if f.is_Add:
+        periods = []
+        for g in f.args:
+            if g.is_Mul:
+                coeff, _ = g.as_independent(symbol)
+                if coeff.is_number and coeff < 0:
+                    raise NotImplementedError("Periods of only certain "
+                                              "trigonometric functions can be "
+                                              "calculated using this method.")
+
+            periods.append(periodicity(g, symbol))
+
+        period = lcm_fraction(periods)
+
+    if period is not None and orig_f.subs(symbol, symbol + period) == orig_f:
+        return period
+
+    else:
+        raise NotImplementedError("Periods of only certain trigonometric functions"
+                                  " can be calculated using this method.")
+
+
+def lcm_fraction(fractions):
+    """Returns the LCM of a list of fractions.
+
+    Examples
+    ========
+    >>> from sympy import S, pi
+    >>> from sympy.calculus.util import lcm_fraction
+    >>> lcm_fraction([S(1)/2, S(3)/4, S(5)/6])
+    15/2
+    >>> lcm_fraction([2*pi, 3*pi, pi, pi/2])
```

Comment:
This was one of the cases where `lcm_list` fails.
I would appreciate some discussion on whether to have this functionality or develop a workaround `lcm_list`
so that it works for us.

Ping @asmeurer @jksuom @hargup @aktech